code/modules/lighting/light_effect_cast.dm

```diff
@@ -0,0 +1,307 @@
+
+/atom/movable/light/proc/cast_light()
+	cast_light_init()
+	cast_main_light()
+	update_light_dir()
+	cast_shadows()
+	update_appearance()
+
+// ----------------------------------------------------------------------------------------------------------------------------------------------------------------
+// -- The shared procs between lights and pesudo-lights.
+
+// Initialisation of the cast_light proc.
+/atom/movable/light/proc/cast_light_init()
+	temp_appearance = list()
+	luminosity = 2*light_range
+
+	//cap light range to 5
+	light_range = min(5, light_range)
+
+	alpha = min(255,max(0,round(light_power*light_power_multiplier*25)))
+
+	if(light_type == LIGHT_SOFT_FLICKER)
+		alpha = initial(alpha)
+		animate(src, alpha = initial(alpha) - rand(30, 60), time = 2, loop = -1, easing = SINE_EASING)
+
+	for (var/mob/M in view(world.view, src))
+		M.check_dark_vision()
+
+	for(var/turf/T in view(2*light_range, src))
+		T.lumcount = -1
+		affecting_turfs |= T
+
+	if(!isturf(loc))
+		for(var/turf/T in affecting_turfs)
+			T.lumcount = -1
+			T.affecting_lights -= src
+		affecting_turfs.Cut()
+		return
+
+/*
+
+Commented out as this doesn't works well with performance currently.
+If you feel like fixing it, try to find a way to calculate the bounds that is less retarded.
+
+/atom/movable/light/shadow/cast_light_init()
+//	. = ..()
+//	if (light_range < 2)
+//		return
+//	if (light_type == LIGHT_DIRECTIONAL)
+//		return
+//
+//	// Basically we check if our cardinals + adjacent tiles are occluded and we adjust our bounds if we are able to.
+//	var/occlusion_north = CheckOcclusion(get_step(loc, NORTH))
+//	var/occlusion_south = CheckOcclusion(get_step(loc, SOUTH))
+//	var/occlusion_east = CheckOcclusion(get_step(loc, EAST))
+//	var/occlusion_west = CheckOcclusion(get_step(loc, WEST))
+//	var/occlusion_northeast = CheckOcclusion(get_step(loc, NORTHEAST))
+//	var/occlusion_northwest = CheckOcclusion(get_step(loc, NORTHWEST))
+//	var/occlusion_southeast = CheckOcclusion(get_step(loc, SOUTHEAST))
+//	var/occlusion_southwest = CheckOcclusion(get_step(loc, SOUTHWEST))
+//
+//	var/visible_top = !(occlusion_north && occlusion_northeast && occlusion_northwest)
+//	var/visible_bottom = !(occlusion_south && occlusion_southeast && occlusion_southwest)
+//	var/visible_left = !(occlusion_east && occlusion_northeast && occlusion_southeast)
+//	var/visible_right = !(occlusion_west && occlusion_southwest && occlusion_northwest)
+//
+//	// If we are visible from the left or right, we have to translate one tile in order for bounds to work
+//	if (visible_left || visible_bottom)
+//		var/vector/V = new(-visible_left, -visible_left)
+//		var/turf/T = loc.get_translated_turf(V)
+//		forceMove(T)
+//		pixel_x = visible_left*WORLD_ICON_SIZE
+//		pixel_y = visible_bottom*WORLD_ICON_SIZE
+//
+//	bound_width = (visible_left + visible_right)*WORLD_ICON_SIZE
+//	bound_height = (visible_top + visible_bottom)*WORLD_ICON_SIZE
+*/
```

Comment:
Still has block comment markers, and as an added bonus, one of the lines of code wouldn't be commented out if the markers weren't there.